src/core/lib/deck.js

```diff
@@ -112,15 +136,9 @@ export default class Deck {
     const {useDevicePixels, autoResizeDrawingBuffer} = props;
     this.animationLoop.setViewParameters({useDevicePixels, autoResizeDrawingBuffer});
     this.stats.timeEnd('deck.setProps');
-  }
 
-  finalize() {
-    this.animationLoop.stop();
-    this.animationLoop = null;
-
-    if (this.layerManager) {
-      this.layerManager.finalize();
-      this.layerManager = null;
+    if (this.controller) {
+      this.controller.setProps(props);
```

Comment:
Need to flatten `viewState`?

Reply:
> Need to flatten viewState?

`viewport-controls.js`, line 144 should handle it?